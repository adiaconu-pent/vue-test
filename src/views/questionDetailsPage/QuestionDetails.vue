<template src="./template.html"></template>
<style lang="scss" src='./style.scss'></style>

<script>
import questionService from '@/services/api/questions'
import { getAlertError, getDateFromInt } from '@/services/commonFunctions'
import ComponentLayout from '@/components/componentLayout/ComponentLayout.vue'
import QuestionBody from '@/components/questionBody/QuestionBody.vue'
import UserDetails from '@/components/userDetails/UserDetails.vue'

export default {
  data () {
    return {
      isLoading: false,
      data: null,
      error: ''
    }
  },
  components: { ComponentLayout, QuestionBody, UserDetails },
  methods: {
    async getQuestionDetails () {
      this.isLoading = true
      questionService
        .getQuestionDetails(this.$route.params.id)
        .then(response => {
          this.data = response.data
        })
        .catch(error => {
          this.error = getAlertError(error)
        })
        .then(() => (this.isLoading = false))
    },
    getDateFromInt
  },
  created () {
    this.getQuestionDetails()
  }
}
</script>
