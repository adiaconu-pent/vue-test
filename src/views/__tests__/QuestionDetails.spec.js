import { shallowMount } from '@vue/test-utils'
import QuestionDetails from '../questionDetailsPage/QuestionDetails'

describe('QuestionDetails.vue', () => {
  test('render the user image', () => {
    const userData = {
      imageUrl: 'https://p7.hiclipart.com/preview/282/256/961/user-profile-avatar-computer-icons-google-account.jpg',
      name: 'Ana',
      score: 10
    }

    const wrapper = shallowMount(QuestionDetails, {
      data: userData
    })
    expect(wrapper).toMatchSnapshot()
  })
})
