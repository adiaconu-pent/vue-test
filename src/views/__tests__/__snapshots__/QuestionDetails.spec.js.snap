// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionDetails.vue render the user image 1`] = `
<v-card>
  <componentlayout-stub error="" isloading="true"></componentlayout-stub>
  <!---->
</v-card>
`;
